tests/cli/main.cpp

```diff
@@ -196,253 +196,188 @@ class client_connection
    std::string wallet_filename;
 };
 
+///////////////////////////////
+// Cli Wallet Fixture
+///////////////////////////////
+
+struct cli_fixture
+{
+    int server_port_number {0};
+    fc::temp_directory app_dir = graphene::utilities::temp_directory_path();
+    std::shared_ptr<graphene::app::application> app1 = start_application(app_dir, server_port_number);
+    client_connection* con = new client_connection(app1, app_dir, server_port_number);
+    std::vector<std::string> nathan_keys { "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3" };
+
+    std::vector<signed_transaction> import_txs;
+    account_object nathan_acct_before_upgrade;
+    signed_transaction upgrade_tx;
+    account_object nathan_acct_after_upgrade;
+    graphene::wallet::brain_key_info bki;
+    signed_transaction create_acct_tx;
+    signed_transaction transfer_tx;
+
+    cli_fixture()
+    {
+        BOOST_TEST_MESSAGE("Setup cli_wallet::boost_fixture_test_case");
+
+        using namespace graphene::chain;
+        using namespace graphene::app;
+
+        try 
+        {
+            BOOST_TEST_MESSAGE("Setting wallet password");
+            con->wallet_api_ptr->set_password("supersecret");
+            con->wallet_api_ptr->unlock("supersecret");
+
+            // import Nathan account
+            BOOST_TEST_MESSAGE("Importing nathan key");
+            BOOST_CHECK_EQUAL(nathan_keys[0], "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3");
+            BOOST_CHECK(con->wallet_api_ptr->import_key("nathan", nathan_keys[0]));
+
+            BOOST_TEST_MESSAGE("Importing nathan's balance");
+            import_txs = con->wallet_api_ptr->import_balance("nathan", nathan_keys, true);
+            nathan_acct_before_upgrade = con->wallet_api_ptr->get_account("nathan");
```

Comment:
The nathan account upgrading should be in a "basic" test case and be invoked by needed test cases, but not in fixture.